qemu/tests/hotplug_virtio_mem.py

```diff
@@ -0,0 +1,52 @@
+import time
+
+from virttest import error_context
+
+from virttest.utils_misc import normalize_data_size
+from virttest.utils_test.qemu import MemoryHotplugTest
+from provider import virtio_mem_utils
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Virtio-mem hotplug test
+    1) Boot guest
+    2) Hotplug virtio-mem device
+    3) Check virtio-mem device
+    4) Resize virtio-mem device twice
+    5) Check virtio-mem device
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    timeout = params.get_numeric("login_timeout", 240)
+    threshold = params.get_numeric("threshold", target_type=float)
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login(timeout=timeout)
+    node_id = params.get_numeric("node_memory_vmem0", 1)
+
+    target_mem = params.get("target_mems")
+    hotplug_test = MemoryHotplugTest(test, params, env)
+    _, vmem_dev = hotplug_test.hotplug_memory(vm, target_mem)
+
+    device_id = vmem_dev.get_qid()
+    requested_size_vmem_test = params.get("requested-size_test_%s" % target_mem)
+    virtio_mem_utils.check_memory_devices(device_id, "0",
+                                          threshold, vm, test)
+    virtio_mem_utils.check_numa_plugged_mem(node_id, "0",
+                                            threshold, vm, test)
```

Comment:
Better avoid the hardcode `0` ,  it is the value of `requested-size_memory_vmem0` (omit unit) in cfg,  or:
```
vmem_dev.get_param('requested-size')
```

Reply:
@yanan-fu updated now, thanks!

Reply:
@yanan-fu updated now, thanks!

Reply:
About the `node-id`,  `initial_req_size`, the current approach all can work,  just it is better to use the same coding style, use the param in cfg or the option value in qemu cli - device params.

Reply:
@yanan-fu 
Aligned `node_id` with the same coding style